jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -613,6 +616,12 @@ public void close() throws IOException
             {
                 // Do an async close
                 channelWrite(content, true, new WriteCompleteCB());
+
+                // If an asynchronous error has happened already we can throw it here.
+                // Otherwise the async call to onWriteComplete must abort if there is a failure as onError may not be called.
+                _abortOnCloseError = true;
+                if (_onError != null)
+                    IO.throwCheckedIOException(_onError);
```

Comment:
This piece of code after `channelWrite()` is in a race with the `WriteCompleteCB`, so I think you should only do the work in `WriteCompleteCB`.

Reply:
This code is about getting the `close()` method to throw if the `WriteCompleteCB` has been failed before we get to this (like if the interceptor failed it). We do this here to report the exception back to the application without aborting the channel so we can send a 500 response.